tests/sentry/workflow_engine/migrations/test_0051_migrate_remaining_issue_alerts.py

```diff
@@ -0,0 +1,402 @@
+        )
+
+        self.issue_alert_missing_matches = self.create_project_rule(
+            name="test2",
+            condition_data=conditions,
+            frequency=5,
+        )
+        data = self.issue_alert_missing_matches.data
+        del data["action_match"]
+        del data["filter_match"]
+        self.issue_alert_missing_matches.update(data=data)
+
+        self.issue_alert_none_matches = self.create_project_rule(
+            name="test3",
+            condition_data=conditions,
+            frequency=5,
+        )
+        data = self.issue_alert_none_matches.data
+        del data["action_match"]
+        del data["filter_match"]
+        self.issue_alert_none_matches.update(data=data)
+
+        self.issue_alert_disabled = self.create_project_rule(
+            name="test4",
+            condition_data=conditions,
+            status=ObjectStatus.DISABLED,
+            frequency=5,
+        )
+
+        self.issue_alert_snoozed = self.create_project_rule(
+            name="test5",
+            condition_data=conditions,
+            frequency=5,
+        )
+        RuleSnooze.objects.create(rule_id=self.issue_alert_snoozed.id)
+
+        self.issue_alert_snoozed_for_user = self.create_project_rule(
+            name="test5-2",
+            condition_data=conditions,
+            frequency=5,
+        )
+        RuleSnooze.objects.create(
+            rule_id=self.issue_alert_snoozed_for_user.id, user_id=self.user.id
+        )
+
+        invalid_conditions = [
+            {
+                "interval": "1h",
+                "id": EventUniqueUserFrequencyConditionWithConditions.id,
+                "value": -1,
+                "comparisonType": "asdf",
+            },
+            {"id": RegressionEventCondition.id},
+        ]
+        self.issue_alert_invalid_condition = self.create_project_rule(
+            name="test6",
+            condition_data=invalid_conditions,
+        )
+
+        self.issue_alert_no_valid_conditions = self.create_project_rule(
+            name="test7",
+            condition_data=[invalid_conditions[0]],
+            frequency=5,
+        )
+
+        self.issue_alert_already_migrated = self.create_project_rule(
+            name="test8",
+            condition_data=conditions,
+            action_match="any",
+            filter_match="any",
+            action_data=self.action_data,
+            frequency=5,
+        )
+        IssueAlertMigrator(self.issue_alert_already_migrated, self.user.id).run()
+
+        self.project2 = self.create_project(organization=self.organization)
+        self.project2_detector = self.create_detector(project=self.project2)
+        self.issue_alert_with_existing_detector = self.create_project_rule(
+            project=self.project2,
+            name="test9",
+            condition_data=conditions,
+            action_data=self.action_data,
+            frequency=5,
+        )
+
+        self.issue_alert_special_case = self.create_project_rule(
```

Comment:
this is the new case we handle

Reply:
i updated the description to explain the missing cases

This issue alert in the test covers both these cases
- Issue alerts whose only conditions were EveryEventConditions (I assume there would only be at most 1)
- Issue occurrence condition with a negative value